api/sdk/product_service.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+package sdk;
+option go_package = "sdk";
+option objc_class_prefix = "GIC";
+
+import "sdk/image_service.proto";
+import "sdk/product.proto";
+
+// CreateProductRequest is used to create a product.
+// The product field is mandatory.
+message CreateProductRequest {
+    // Product contains all information needed to create a product.
+    Product product = 1;
+
+    // Image is the image that are going to be assigned to this product.
+    CreateImageRequest image = 2;
+}
+
+// CreateProductResponse confirms the creation of a product.
+message CreateProductResponse {
+    // Product is the created product.
+    ProductWithUniqueId product = 1;
```

Comment:
Really like the composition pattern in this demo. This will likely stutter as clients will need to write `response.product.product.name` to get to the underlying product. We haven't really found a better naming scheme here though, but it might be worth pointing out in the talk.